test/services/AggregatorSpec.scala

```diff
@@ -63,7 +63,186 @@ class RowAggregatorSpec extends BaseSpec {
     val instructionGroup: InstructionGroup = InstructionGroup(Seq(instruction, instruction1, instruction2))
   }
 
-  "Row aggregator" must {
+  private trait NumberedListTest extends Test {
+    val num1Phrase = Phrase(Vector("Line1", "Welsh Line1"))
+    val num2Phrase = Phrase(Vector("Line1", "Welsh Line1"))
+    val num3Phrase = Phrase(Vector("Line1", "Welsh Line1"))
+    val num4Phrase = Phrase(Vector("Line1", "Welsh Line1"))
+
+    val num1ListCo = NumListCallout(num1Phrase, Seq(""), false)
+    val num2ListCo = NumListCallout(num2Phrase, Seq(""), true)
+    val num3ListCo = NumListCallout(num3Phrase, Seq(""), true)
+    val num4ListCo = NumListCallout(num4Phrase, Seq(""), true)
+    val num1CircListCo = NumCircListCallout(num1Phrase, Seq(""), false)
+    val num2CircListCo = NumCircListCallout(num2Phrase, Seq(""), true)
+    val num3CircListCo = NumCircListCallout(num3Phrase, Seq(""), true)
+    val num4CircListCo = NumCircListCallout(num4Phrase, Seq(""), true)
+  }
+
+  "NumberedList aggregation" must {
+
+    "add an isolated number list co with stack equals false into a row group of size 1" in new NumberedListTest {
+      val stanzas: Seq[VisualStanza] =
+        Seq(
+          callout,
+          num1ListCo,
+          instruction)
+
+      val aggregatedStanzas: Seq[Stanza] = Aggregator.aggregateStanzas(Nil)(stanzas)
+      aggregatedStanzas(1) shouldBe NumListGroup(Seq(num1ListCo))
+    }
+
+  "add an isolated number list co with stack equals true into a row group of size 1" in new NumberedListTest {
+    val stanzas: Seq[VisualStanza] = Seq(
+      callout,
+      num1ListCo,
+      instruction
+    )
+
+    val aggregatedStanzas: Seq[Stanza] = Aggregator.aggregateStanzas(Nil)(stanzas)
+
+    aggregatedStanzas(1) shouldBe NumListGroup(Seq(num1ListCo))
+  }
+
+  "create two number list groups for two contiguous rows with stack set to false" in new NumberedListTest {
+
+    val stanzas: Seq[VisualStanza] = Seq(
+      callout,
+      num1ListCo,
+      num1ListCo,
```

Comment:
Should this be num2ListCo?

Reply:
actually test later with num2ListCo so this test just reuses num1ListCo as only one with stacked false - so ignore comment above :)

Reply:
ok